File: fix-ts.js
const fs = require('fs');
const file = 'src/app/admin/products/page.tsx';
let content = fs.readFileSync(file, 'utf8');

content = content.replace(
    /const escapeStr = \(str\) => {/g,
    'const escapeStr = (str: any) => {'
);

fs.writeFileSync(file, content, 'utf8');
console.log('Fixed typescript error');

File: fix_ts.py
with open("src/app/api/orders/[id]/route.ts", "r", encoding="utf-8") as f:
    content = f.read()

old_cancel = """await adjustInventory(order.items.map(l => ({ productId: l.productId, size: l.size, quantity: l.quantity })), true);"""
new_cancel = """await adjustInventory(order.items.map(l => ({ productId: l.productId, size: l.size || undefined, quantity: l.quantity })), true);"""
content = content.replace(old_cancel, new_cancel)

with open("src/app/api/orders/[id]/route.ts", "w", encoding="utf-8") as f:
    f.write(content)
print("Fixed TS error in route.ts")
